controllers/provider-azure/pkg/webhook/controlplane/ensurer.go

```diff
@@ -229,3 +238,65 @@ func (e *ensurer) EnsureKubeletCloudProviderConfig(ctx context.Context, ectx gen
 	*data = cm.Data[azure.CloudProviderConfigMapKey]
 	return nil
 }
+
+// EnsureAdditionalFile ensures additional systemd files
+func (e *ensurer) EnsureAdditionalFiles(ctx context.Context, ectx genericmutator.EnsurerContext, files *[]extensionsv1alpha1.File) error {
+	if err := e.ensureAcrConfigFile(ctx, ectx, files); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e *ensurer) ensureAcrConfigFile(ctx context.Context, ectx genericmutator.EnsurerContext, files *[]extensionsv1alpha1.File) error {
+	for _, f := range *files {
+		if f.Path == acrConfigPath {
+			return nil
```

Comment:
This means that, when the user for example exchanges the identity, it will never be updated again because it already exists. I think a unit test for this case makes sense as well.